BrowserKit/Sources/SiteImageView/ImageProcessing/SiteImageFetcher/SiteImageDownloader.swift

```diff
@@ -15,13 +15,25 @@ protocol SiteImageDownloader {
         with url: URL,
         completionHandler: ((Result<SiteImageLoadingResult, KingfisherError>) -> Void)?
     ) -> DownloadTask?
+
+    func downloadImage(with url: URL) async -> Result<SiteImageLoadingResult, KingfisherError>
+}
+
+extension SiteImageDownloader {
+    func downloadImage(with url: URL) async -> Result<SiteImageLoadingResult, KingfisherError> {
+        await withCheckedContinuation { continuation in
+            _ = downloadImage(with: url) { result in
+                continuation.resume(returning: result)
+            }
+        }
+    }
```

Comment:
One of the main points of async/await is that you don't want to use `Result`, you'd want to actually have the function throw. Async/await is meant to be a more functional pattern, which means all exits out of the function would be explicit and the return type would be singular. So this should be converted into a throwing function a la
```suggestion
    func downloadImage(with url: URL) async throws -> SiteImageLoadingResult {
        await withCheckedThrowingContinuation { continuation in
            _ = downloadImage(with: url) { result in
            // inside here, you'd actually check the result and throw on an error, or resume or success
                continuation.resume(returning: result)
            }
        }
    }
```